### PR TITLE
Rescuing Mouf P and Pet Penguin Quests

### PR DESCRIPTION
**Pull request by sLackeR-hub**
**Completed Quests**

**Updated: (To Accept Items):**
Elijah
Larinne Kerendova
Mr. P

**Added:**
Mr. P
Hoarfrost
Letter from Uber P
Uber P
Mouf
Letter from Mouf
Letter from Mouf and Uber P
Baby Penguin
Penguin Guardian 
Cowardly Snowman
Tiny's Head
Door
Lever
Tiny
Pillar of Frost
Wall of Ice
Frigid Grotto
Frigid Ice Golem
Frigid Mist Golem
Penguin Guard
Surface
Security Station
Security Door
Pet Penguin
Pet Penguin Igloo
Marked Candy Wrapper
Small Sack of Penguin Treats
Penguin Nest
Especially Famished Eater
Penguin Egg
Partially-eaten Cookie
Portable Igloo

**Added to Landblocks:**
Uber P
Mr. P
Frigid Grotto Portal

**Updated Landblock:**
37E4

### CODE REVIEW

**Review by ziang4891 — CHANGES_REQUESTED** [review]
init_Level values in weenie_properties_attribute_2nd table need to be fixed as described in content_recreation Discord channel

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Human/33939 Larinne Kerendova.sql [hunk 1]
```diff
@@ -0,0 +1,302 @@
+DELETE FROM `weenie` WHERE `class_Id` = 33939;
+
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Human/33939 Larinne Kerendova.sql:2]: If you bring a weenie file forward in patches, the original should be removed; this NPC also exists in 2006-09-ComeWhatFollows patch folder

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Penguin/46299 Tiny.sql [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+INSERT INTO `weenie_properties_create_list` (`object_Id`, `destination_Type`, `weenie_Class_Id`, `stack_Size`, `palette`, `shade`, `try_To_Bond`)
+VALUES (46299, 9, 20252,  0, 0, 0, False) /* Create Scroll of Belly of Lead (20252) for ContainTreasure */
+     , (46299, 9, 46288,  0, 0, 0, False) /* Create Tiny's Head (46288) for ContainTreasure */
+     , (46299, 9, 49534,  0, 0, 0, False) /* Create Fire Phyntos Wasp Essence (125) (49534) for ContainTreasure */
+     , (46299, 9,   359,  0, 0, 0, False) /* Create War Hammer (359) for ContainTreasure */
+     , (46299, 9, 25643,  0, 0, 0, False) /* Create Leather Girth (25643) for ContainTreasure */
+     , (46299, 9, 31864,  0, 0, 0, False) /* Create Teardrop Crown (31864) for ContainTreasure */
+     , (46299, 9, 45117,  0, 0, 0, False) /* Create Frost Hammer (45117) for ContainTreasure */
+     , (46299, 9,  7795,  0, 0, 0, False) /* Create Frost Naginata (7795) for ContainTreasure */
+     , (46299, 9,   133,  0, 0, 0, False) /* Create Slippers (133) for ContainTreasure */;
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Penguin/46299 Tiny.sql:139]: This table should be condensed down to just the Tiny's Head entry, as the other items were just a snapshot of the random loot at the time of capture, which DeathTreasureType should cover.  Also, for Tiny's Head, shade would be set to 1, as that data point is actually treated as probability of dropping for ContainTreasure.  A zero would mean it doesn't drop at all.

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+INSERT INTO `weenie_properties_d_i_d` (`object_Id`, `type`, `value`)
+VALUES (46278,   1,   33556222) /* Setup */
+     , (46278,   2,  150995088) /* MotionTable */
+     , (46278,   3,  536871000) /* SoundTable */
+     , (46278,   4,  805306406) /* CombatTable */
+     , (46278,   8,  100669125) /* Icon */
+     , (46278,  22,  872415346) /* PhysicsEffectTable */
+     , (46278,  35,       1003) /* DeathTreasureType - Loot Tier: 7 */;
+
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql:76]: A level 105 mob is not likely to have tier 7 loot

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+VALUES (46278,  6, 0, 3, 0,  70, 0, 0) /* MeleeDefense        Specialized */
+     , (46278,  7, 0, 3, 0, 103, 0, 0) /* MissileDefense      Specialized */
+     , (46278, 47, 0, 3, 0,  70, 0, 0) /* MissileWeapons      Specialized */
+     , (46278, 45, 0, 3, 0, 110, 0, 0) /* LightWeapons        Specialized */
+     , (46278, 14, 0, 3, 0,  70, 0, 0) /* ArcaneLore          Specialized */
+     , (46278, 15, 0, 3, 0,  58, 0, 0) /* MagicDefense        Specialized */
+     , (46278, 20, 0, 3, 0, 100, 0, 0) /* Deception           Specialized */
+     , (46278, 24, 0, 3, 0,  80, 0, 0) /* Run                 Specialized */
+     , (46278, 12, 0, 3, 0,  56, 0, 0) /* ThrownWeapon		  Specialized */;
+
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql:100]: ThrownWeapon skill is covered by MissileWeapons, so doesn't need to be added

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46303 Wall of Ice.sql [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (46303,   1, True ) /* Stuck */
+     , (46303,   6, False) /* AiUsesMana */
+     , (46303,  11, False) /* IgnoreCollisions */
+     , (46303,  12, True ) /* ReportCollisions */
+     , (46303,  13, False) /* Ethereal */
+     , (46303,  14, True ) /* GravityStatus */
+     , (46303,  19, True ) /* Attackable */
+     , (46303,  29, True ) /* NoCorpse */;
+
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46303 Wall of Ice.sql:23]: Missing NpcLooksLikeObject and a CombatTable entry in weenie_properties_d_i_d.  Can probably use the one from https://github.com/ACEmulator/ACE-World-16PY-Patches/blob/master/Database/Patches/2005-12-ThroughSacrificeStrength/9%20WeenieDefaults/Creature/Unsorted/80027%20Mana%20Barrier.sql

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46320 Security Station.sql [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+     , (46320,  16,         32) /* ItemUseable - Remote */
+     , (46320,  81,          1) /* MaxGeneratedObjects */
+     , (46320,  82,          0) /* InitGeneratedObjects */
+     , (46320,  83,       2048) /* ActivationResponse - Emote */
+     , (46320,  93,    6292504) /* PhysicsState - ReportCollisions, IgnoreCollisions, Gravity, ReportCollisionsAsEnvironment, EdgeSlide */
+     , (46320,  95,          3) /* RadarBlipColor - White */;
+
+INSERT INTO `weenie_properties_bool` (`object_Id`, `type`, `value`)
+VALUES (46320,   1, True ) /* Stuck */
+     , (46320,  19, False) /* Attackable */;
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46320 Security Station.sql:19]: Missing NpcLooksLikeObject property

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Portal/46308 Surface.sql [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+INSERT INTO `weenie_properties_float` (`object_Id`, `type`, `value`)
+VALUES (46308,  54,    -0.1) /* UseRadius */;
+
+INSERT INTO `weenie_properties_string` (`object_Id`, `type`, `value`)
+VALUES (46308,   1, 'Surface') /* Name */;
+
+INSERT INTO `weenie_properties_d_i_d` (`object_Id`, `type`, `value`)
+VALUES (46308,   1,   33554867) /* Setup */
+     , (46308,   2,  150994947) /* MotionTable */
+     , (46308,   8,  100667499) /* Icon */;
```
Comment (ziang4891) [inline Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Portal/46308 Surface.sql:26]: Missing weenie_properties_position Destination

**Review by ziang4891 — CHANGES_REQUESTED** [review]
Weenies with ClothingBase should probably have an associated PaletteTemplate INT property, which can usually be found by searching for the wcid on Optim's site (http://ac.yotesfan.com/)

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/42338 Hoarfrost.sql [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+     , (42338,  1091,  2.008)  /* Fire Protection Self III */
+     , (42338,  1158,  2.013)  /* Heal Self III */
+     , (42338,  1239,  2.008)  /* Drain Health Other III */
+     , (42338,  1324,  2.017)  /* Imperil Other III */
+     , (42338,  1340,  2.008)  /* Weakness Other III */
+     , (42338,  1417,  2.008)  /* Slowness Other III */
+     , (42338,  1810,  2.004)  /* Frost Streak III */;
+
+INSERT INTO `weenie_properties_create_list` (`object_Id`, `destination_Type`, `weenie_Class_Id`, `stack_Size`, `palette`, `shade`, `try_To_Bond`)
+VALUES (42338, 9,  6876,  0, 0, 0.001, False) /* Create Sturdy Iron Key (6876) for ContainTreasure */;
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/42338 Hoarfrost.sql]: Fixed Hoarfrost completely. Going through all other critters.

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+     , (46300,    97,   2.05) /* Whirling Blade VI */
+     , (46300,   176,   2.05) /* Fester Other VI */
+     , (46300,   526,   2.05) /* Acid Vulnerability Other VI */
+     , (46300,  1053,   2.03) /* Bludgeoning Vulnerability Other VI */
+     , (46300,  1065,   2.02) /* Cold Vulnerability Other VI */
+     , (46300,  1089,   2.02) /* Lightning Vulnerability Other VI */
+     , (46300,  1108,   2.07) /* Fire Vulnerability Other VI */
+     , (46300,  1132,   2.05) /* Blade Vulnerability Other VI */
+     , (46300,  1156,    2.1) /* Piercing Vulnerability Other VI */
+     , (46300,  3878,      3) /* Incendiary Strike */;
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql]: Fixed

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+     , (46300,  5,  4,1000, 0.75,  500,  250,  250,  250,  250,  250,  250,  250,    0, 2,    0,  0.2,    0,    0,  0.2,    0,    0,  0.2,    0,    0,  0.2,    0) /* Hand - Bludgeon */
+     , (46300,  6,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0, 0.13, 0.18,    0, 0.13, 0.18,    0, 0.13, 0.18,    0, 0.13, 0.18) /* UpperLeg */
+     , (46300,  7,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0,    0,  0.6,    0,    0,  0.6,    0,    0,  0.6,    0,    0,  0.6) /* LowerLeg */
+     , (46300,  8,  4,1000, 0.75,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0,    0, 0.22,    0,    0, 0.22,    0,    0, 0.22,    0,    0, 0.22) /* Foot - Bludgeon */;
+
+INSERT INTO `weenie_properties_position` (`object_Id`, `position_Type`, `obj_Cell_Id`, `origin_X`, `origin_Y`, `origin_Z`, `angles_W`, `angles_X`, `angles_Y`, `angles_Z`)
+VALUES (46300, 8040, 1481441546, 100.0045, -59.59311, -24.00334, 0.7970368, 0, 0, -0.6039308) /* PCAPRecordedLocation */
+/* @teleloc 0x584D010A [100.004500 -59.593110 -24.003340] 0.797037 0.000000 0.000000 -0.603931 */;
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql]: Removed

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+     , (46300,  3,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 1, 0.23, 0.03,    0, 0.23, 0.03,    0, 0.23, 0.03,    0, 0.23, 0.03,    0) /* UpperArm */
+     , (46300,  4,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 2,    0,  0.3,    0,    0,  0.3,    0,    0,  0.3,    0,    0,  0.3,    0) /* LowerArm */
+     , (46300,  5,  4,1000, 0.75,  500,  250,  250,  250,  250,  250,  250,  250,    0, 2,    0,  0.2,    0,    0,  0.2,    0,    0,  0.2,    0,    0,  0.2,    0) /* Hand - Bludgeon */
+     , (46300,  6,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0, 0.13, 0.18,    0, 0.13, 0.18,    0, 0.13, 0.18,    0, 0.13, 0.18) /* UpperLeg */
+     , (46300,  7,  4,  0,    0,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0,    0,  0.6,    0,    0,  0.6,    0,    0,  0.6,    0,    0,  0.6) /* LowerLeg */
+     , (46300,  8,  4,1000, 0.75,  500,  250,  250,  250,  250,  250,  250,  250,    0, 3,    0,    0, 0.22,    0,    0, 0.22,    0,    0, 0.22,    0,    0, 0.22) /* Foot - Bludgeon */;
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Elemental/46300 Pillar of Frost.sql]: Changed to 10.

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46303 Wall of Ice.sql (hunk 5 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46303 Wall of Ice.sql:24]: Fixed.

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Golem/46305 Frigid Ice Golem.sql [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+     , (46305, 14, 0, 2, 0, 180, 0, 0) /* ArcaneLore          Trained */
+     , (46305, 15, 0, 3, 0, 175, 0, 0) /* MagicDefense        Specialized */
+     , (46305, 20, 0, 2, 0, 100, 0, 0) /* Deception           Trained */
+     , (46305, 22, 0, 2, 0,  10, 0, 0) /* Jump                Trained */
+     , (46305, 24, 0, 2, 0,  10, 0, 0) /* Run                 Trained */
+     , (46305, 31, 0, 3, 0, 150, 0, 0) /* CreatureEnchantment Specialized */
+     , (46305, 32, 0, 3, 0, 150, 0, 0) /* ItemEnchantment     Specialized */
+     , (46305, 33, 0, 3, 0, 150, 0, 0) /* LifeMagic           Specialized */
+     , (46305, 34, 0, 3, 0, 150, 0, 0) /* WarMagic            Specialized */;
+
```
[the remark this answers is not in the document]
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Golem/46305 Frigid Ice Golem.sql:112]: Fixed.

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql (hunk 3 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql:76]: Changed to 6

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql (hunk 4 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Snowman/46278 Cowardly Snowman.sql:99]: Removed/

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46320 Security Station.sql (hunk 6 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Creature/Unsorted/46320 Security Station.sql]: Added

File: Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Portal/46308 Surface.sql (hunk 7 above)
Comment (sLackeR-hub) [reply Database/Patches/2012-07-SoullessSisters/9 WeenieDefaults/Portal/46308 Surface.sql:26]: Fixed.

**Review by ziang4891 — APPROVED** [review]

**Review by DridsGhost — APPROVED** [review]

**Pull request merged**